examples/map/map_segment.py

```diff
@@ -0,0 +1,113 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+
+A demonstration extracting a region of a particular map based on
+world coordinates in different systems.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import NorthOffsetFrame
+from sunpy.data.sample import AIA_171_IMAGE
+
+######################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+
+######################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel. From this we then transform the coordinates to the required frame.
+# For this example we are going to extract a region based on the
+# heliographic Stonyhurst coordinates, so we transform to that frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+######################################################################
+# Let's then segment the data based on coordinates and create a boolean mask
+# where True indicates invalid or deselected data.
+# This is the convention used by the `numpy.ma` module.
+segment = np.logical_or(
+    all_hgs.lon >= 50 * u.deg,
+    all_hgs.lon <= 0 * u.deg)
```

Comment:
We can combine all this logic into the one statement.
```suggestion
segment = np.logical_or(
    all_hgs.lon >= 50 * u.deg,
    all_hgs.lon <= 0 * u.deg,
    ~np.isnan(all_hgs.lon))
```

Alternatively `segment = np.logical_or(segment, np.logical_not(np.isnan(all_hgs.lon)))` is more explicit than the multiplication below.

Reply:
This doesn't produce the right plot though

Reply:
This still produces the issue where the `NaN` values aren't masked
![image](https://user-images.githubusercontent.com/50923538/112461861-63892b00-8d86-11eb-9b5d-ad09ca23fe2e.png)

Reply:
wait well then use `np.isnan`

Reply:
Heh, I suppose `np.inf` can't be a possibility, so sure.

Reply:
Oh yeah, forgot that existed.